AniHyou/Screens/MediaList/Views/MediaListView.swift

```diff
@@ -122,6 +69,79 @@ struct MediaListView: View {
         }
         .navigationTitle(viewModel.mediaListStatus.localizedName)
     }
+    
+    func buildListItem(item: UserMediaListQuery.Data.Page.MediaList!) -> some View {
+        NavigationLink(destination: MediaDetailsView(mediaId: item.mediaId)) {
+            switch listStyle {
+            case 1:
+                MediaListItemMinimalView(item: item)
+            case 2:
+                MediaListItemCompactView(item: item)
+            default:
+                MediaListItemStandardView(item: item)
+            }
+        }
+        .swipeActions(edge: .leading) {
+            if incrementLongSwipeDirection == .right {
+                if shouldShowIncrementButton() {
+                    Button(action: {
+                        updateEntryProgress(item: item)
+                    }) {
+                        if type == .anime {
+                            Label("Ep", systemImage: "plus")
+                        // should show a sheet to add a rating
+                        } else if type == .manga {
+                            Label("Ch", systemImage: "plus")
+                        }
+                    }
+                    .tint(.green)
+                }
+            }
+        }
+        .swipeActions(edge: .trailing) {
+            if incrementLongSwipeDirection == .left {
+                if shouldShowIncrementButton() {
+                    Button(action: {
+                        updateEntryProgress(item: item)
+                    }) {
+                        if type == .anime {
+                            Label("Ep", systemImage: "plus")
+                        // should show a sheet to add a rating
+                        } else if type == .manga {
+                            Label("Ch", systemImage: "plus")
+                        }
+                    }
+                    .tint(.green)
+                }
+            }
+            Button(action: {
+                viewModel.selectedItem = item
+                showingEditSheet = true
+            }) {
+                Label("Edit", systemImage: "square.and.pencil")
+            }
+            .tint(.blue)
+        }
+    }
+    
+    func updateEntryProgress(item: UserMediaListQuery.Data.Page.MediaList!) {
+        var status: GraphQLEnum<MediaListStatus>
+        if item.status == .planning {
+            status = .case(.current)
+        } else {
+            status = item.status!
+        }
+        viewModel.updateEntryProgress(entryId: item.id, progress: item.progress! + 1, status: status)
+    }
+    
+    func shouldShowIncrementButton() -> Bool {
```

Comment:
I think a calculated var would be better here, but you can leave as it

Reply:
ahh you're right... javascript roots are showing 😩